packages/html/src/client.ts

```diff
@@ -0,0 +1,35 @@
+import type { ConfigCtrlState } from '@web3modal/core'
+import { ClientCtrl, ConfigCtrl, ModalCtrl } from '@web3modal/core'
+import type { EthereumClient } from '@web3modal/ethereum'
+
+/**
+ * Types
+ */
+type Web3ModalConfig = Omit<ConfigCtrlState, 'enableStandaloneMode' | 'standaloneChains'>
+
+/**
+ * Client
+ */
+export class Web3Modal {
+  public constructor(config: Web3ModalConfig, client: EthereumClient) {
```

Comment:
I don't quite understand the vision with this.

Are we going to make a generic `Client` type? With the imminent integration of Solib, how do we plan to make this more generic?

This is absolutely not a blocker for this PR, just want to know

Reply:
As you said, this will become a generalised client type in the future that all clients should satisfy.